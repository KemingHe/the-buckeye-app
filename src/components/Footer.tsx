"use client";

export default function Footer(): JSX.Element {
  return (
    <footer className="footer footer-center bg-base-300 text-base-content p-5">
      <aside>
        <p className="flex space-x-2">
          <span>Keming He</span>
          <span aria-hidden={true}>&#x2022;</span>
          <span>&#169; {new Date().getFullYear()}</span>
        </p>
      </aside>
    </footer>
  );
}
