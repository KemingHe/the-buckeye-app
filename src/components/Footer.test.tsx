import { type RenderResult, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import Footer from "./Footer";

describe("Footer component", () => {
  it("renders the footer with the correct content", () => {
    render(<Footer />);

    // Select the footer by its role.
    const footerElement = screen.getByRole("contentinfo");
    expect(footerElement).toBeVisible();

    // Select the text content.
    const authorContent: string = "Keming He";
    const copyRightContent: string = `Â© ${new Date().getFullYear()}`;
    expect(screen.getByText(authorContent)).toBeVisible();
    expect(screen.getByText(copyRightContent)).toBeVisible();
  });

  it("matches the snapshot", () => {
    const { asFragment }: RenderResult = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
