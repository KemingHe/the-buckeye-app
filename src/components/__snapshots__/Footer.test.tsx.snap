// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer footer-center bg-base-300 text-base-content p-5"
  >
    <aside>
      <p
        class="flex space-x-2"
      >
        <span>
          Keming He
        </span>
        <span
          aria-hidden="true"
        >
          •
        </span>
        <span>
          © 2024
        </span>
      </p>
    </aside>
  </footer>
</DocumentFragment>
`;
